src/devSVG.cpp

```diff
@@ -124,6 +131,37 @@ inline void write_attrs_linetype(FILE* f, int lty, double lwd, int col) {
     fputs("'", f);
     break;
   }
+
+  // Set line end shape
+  switch(lend)
+  {
+  case GE_ROUND_CAP:
+    write_attr_str(f, "stroke-linecap", "round");
+    break;
+  case GE_SQUARE_CAP:
+    write_attr_str(f, "stroke-linecap", "square");
+    break;
+  case GE_BUTT_CAP: // doing nothing, default is butt
+  default:
+    break;
+  }
+
+  // Set line join shape
+  switch(ljoin)
+  {
+  case GE_ROUND_JOIN:
+    write_attr_str(f, "stroke-linejoin", "round");
+    break;
+  case GE_BEVEL_JOIN:
+    write_attr_str(f, "stroke-linejoin", "bevel");
+    break;
+  case GE_MITRE_JOIN: // We don't need to write "stroke-linejoin" attribute here,
+                      // since the default is "miter". However, we do need to
+                      // specify "stroke-miterlimit".
+    write_attr_dbl(f, "stroke-miterlimit", lmitre);
```

Comment:
Actually there is a mismatch of defaults between R and SVG standard.

- Line end: R default is "round" while SVG default is "butt".
- Line join: R default is "round" and SVG default is "miter".
- Miter limit, R default is 10 while SVG default is 4.

This means that R defaults are never SVG defaults and if we strictly conform to the standard, we have to generate some extra text.